### ISSUE TITLE
Access denied to Docker.io

### ISSUE THREAD

**Issue by cmartin616** [crazy-max/diun#98]
#### Steps to reproduce this issue

1. Docker container running diun.
2.  Was working until June 16th.

#### Expected behaviour

Diun should notify when a new image is found

#### Actual behaviour

Access denied to Docker.io endpoints:

Thu, 18 Jun 2020 16:12:05 EDT DBG app/internal/logging/logger.go:68 > [containers/image] Accessing "docker.io/library/sha256:445909f0eaa51a79a49b1c4ae3f8b461cc4981df768f11d9f25c74573fe86f12" failed: Error reading manifest 445909f0eaa51a79a49b1c4ae3f8b461cc4981df768f11d9f25c74573fe86f12 in docker.io/library/sha256: errors:,
denied: requested access to the resource is denied,
unauthorized: authentication required,
,
Thu, 18 Jun 2020 16:12:05 EDT ERR app/internal/app/diun.go:116 > Cannot run job error="Error reading manifest 445909f0eaa51a79a49b1c4ae3f8b461cc4981df768f11d9f25c74573fe86f12 in docker.io/library/sha256: errors:\ndenied: requested access to the resource is denied\nunauthorized: authentication required\n" provider=docker,

### Configuration

* Diun version :  latest Docker image
* Platform (windows/linux) : Ubuntu Server 18.04
* System info (type `uname -a`) : Linux [machine name] 4.15.0-106-generic #107-Ubuntu SMP Thu Jun 4 11:27:52 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux

```yaml
db:
  path: /data/diun.db

watch:
  workers: 10
  schedule: "0 * * * *"
  first_check_notif: false

notif:
  telegram:
    token: [token]
    chat_ids:
      - [chatid]

providers:
  docker:
     watch_stopped: true
     watch_by_default: true

```

### Logs

```
Thu, 18 Jun 2020 16:12:04 EDT DBG app/internal/logging/logger.go:68 > [containers/image] GET https://auth.docker.io/token?scope=repository%3Aportainer%2Fportainer%3Apull&service=registry.docker.io,
Thu, 18 Jun 2020 16:12:04 EDT DBG app/internal/logging/logger.go:68 > [containers/image] Ping https://registry-1.docker.io/v2/ status 401,
Thu, 18 Jun 2020 16:12:04 EDT DBG app/internal/logging/logger.go:68 > [containers/image] GET https://auth.docker.io/token?scope=repository%3Anodered%2Fnode-red-docker%3Apull&service=registry.docker.io,
Thu, 18 Jun 2020 16:12:04 EDT DBG app/internal/logging/logger.go:68 > [containers/image] GET https://registry-1.docker.io/v2/linuxserver/sonarr/manifests/5.14,
Thu, 18 Jun 2020 16:12:04 EDT DBG app/internal/logging/logger.go:68 > [containers/image] GET https://registry-1.docker.io/v2/linuxserver/tautulli/manifests/latest,
Thu, 18 Jun 2020 16:12:04 EDT DBG app/internal/logging/logger.go:68 > [containers/image] GET https://registry-1.docker.io/v2/portainer/portainer/manifests/latest,
Thu, 18 Jun 2020 16:12:04 EDT DBG app/internal/logging/logger.go:68 > [containers/image] GET https://registry-1.docker.io/v2/nodered/node-red-docker/manifests/latest,
Thu, 18 Jun 2020 16:12:05 EDT DBG app/internal/logging/logger.go:68 > [containers/image] GET https://registry-1.docker.io/v2/linuxserver/radarr/manifests/sha256:8aef7211053bceb3a563f92959fb525a709a38ae82a435963767566ec6f99a1a,
Thu, 18 Jun 2020 16:12:05 EDT DBG app/internal/logging/logger.go:68 > [containers/image] GET https://registry-1.docker.io/v2/linuxserver/piwigo/manifests/sha256:659439ab01f7e3dd624040d9941b73ede3391143a7c3fb8411f5942e6042454f,
Thu, 18 Jun 2020 16:12:05 EDT DBG app/internal/logging/logger.go:68 > [containers/image] Downloading /v2/linuxserver/lazylibrarian/blobs/sha256:558e2c730aeafa64038434ff4df76c63f5784cb03cd6b02d25b4c97d79c60604,
Thu, 18 Jun 2020 16:12:05 EDT DBG app/internal/logging/logger.go:68 > [containers/image] GET https://registry-1.docker.io/v2/linuxserver/lazylibrarian/blobs/sha256:558e2c730aeafa64038434ff4df76c63f5784cb03cd6b02d25b4c97d79c60604,
Thu, 18 Jun 2020 16:12:05 EDT DBG app/internal/logging/logger.go:68 > [containers/image] GET https://registry-1.docker.io/v2/linuxserver/ombi/manifests/sha256:d0d63060fc41dfd7baedeb4eaf6c4355d124ad69eeaf230c0a8bdc7f54795591,
Thu, 18 Jun 2020 16:12:05 EDT DBG app/internal/logging/logger.go:68 > [containers/image] GET https://registry-1.docker.io/v2/linuxserver/sabnzbd/manifests/sha256:87a9b891ba2342ddfaa1a16c66ebe67202f05be89a7b754ede3e07928e9349f4,
Thu, 18 Jun 2020 16:12:05 EDT DBG app/internal/logging/logger.go:68 > [containers/image] GET https://registry-1.docker.io/v2/linuxserver/tautulli/manifests/sha256:1c01c1d1c759b561ac0385d7c18dbce4189ea7d485652ae9fae0d867f08a1b58,
Thu, 18 Jun 2020 16:12:05 EDT DBG app/internal/logging/logger.go:68 > [containers/image] Downloading /v2/linuxserver/letsencrypt/blobs/sha256:49815c2a35f92c1c64e363849d3e920d6cd471969904ce336a2f7cac58a8feca,
Thu, 18 Jun 2020 16:12:05 EDT DBG app/internal/logging/logger.go:68 > [containers/image] GET https://registry-1.docker.io/v2/linuxserver/letsencrypt/blobs/sha256:49815c2a35f92c1c64e363849d3e920d6cd471969904ce336a2f7cac58a8feca,
Thu, 18 Jun 2020 16:12:05 EDT DBG app/internal/logging/logger.go:68 > [containers/image] GET https://registry-1.docker.io/v2/portainer/portainer/manifests/sha256:e879687781d32f7c10440f7b5f67e5e7f2399179af6196c66c76182758b76299,
Thu, 18 Jun 2020 16:12:05 EDT DBG app/internal/logging/logger.go:68 > [containers/image] Downloading /v2/nodered/node-red-docker/blobs/sha256:0bd9a1d2200474e7471bada2eb633f7193320ee47cb3b8aa34326d19f7f485c6,
Thu, 18 Jun 2020 16:12:05 EDT DBG app/internal/logging/logger.go:68 > [containers/image] GET https://registry-1.docker.io/v2/nodered/node-red-docker/blobs/sha256:0bd9a1d2200474e7471bada2eb633f7193320ee47cb3b8aa34326d19f7f485c6,
Thu, 18 Jun 2020 16:12:05 EDT DBG app/internal/logging/logger.go:68 > [containers/image] GET https://registry-1.docker.io/v2/linuxserver/sonarr/manifests/sha256:12b88673e7fbdc23d4b2b6bd3bffdd7070f42dd26a17bb2ae339312b9bdfa235,
Thu, 18 Jun 2020 16:12:05 EDT DBG app/internal/app/job.go:167 > No changes image=docker.io/linuxserver/lazylibrarian:latest provider=docker,
Thu, 18 Jun 2020 16:12:05 EDT DBG app/internal/logging/logger.go:68 > [containers/image] Trying to access "docker.io/sgtsquiggs/deluge-openvpn:latest",
Thu, 18 Jun 2020 16:12:05 EDT DBG app/internal/logging/logger.go:68 > [containers/image] Returning credentials from DockerAuthConfig,
Thu, 18 Jun 2020 16:12:05 EDT DBG app/internal/logging/logger.go:68 > [containers/image] Using registries.d directory /etc/containers/registries.d for sigstore configuration,
Thu, 18 Jun 2020 16:12:05 EDT DBG app/internal/logging/logger.go:68 > [containers/image]  No signature storage configuration found for docker.io/sgtsquiggs/deluge-openvpn:latest,
Thu, 18 Jun 2020 16:12:05 EDT DBG app/internal/logging/logger.go:68 > [containers/image] Looking for TLS certificates and private keys in /etc/docker/certs.d/docker.io,
Thu, 18 Jun 2020 16:12:05 EDT DBG app/internal/logging/logger.go:68 > [containers/image] GET https://registry-1.docker.io/v2/,
Thu, 18 Jun 2020 16:12:05 EDT DBG app/internal/logging/logger.go:68 > [containers/image] Ping https://registry-1.docker.io/v2/ status 401,
Thu, 18 Jun 2020 16:12:05 EDT DBG app/internal/logging/logger.go:68 > [containers/image] GET https://auth.docker.io/token?scope=repository%3Asgtsquiggs%2Fdeluge-openvpn%3Apull&service=registry.docker.io,
Thu, 18 Jun 2020 16:12:05 EDT DBG app/internal/app/job.go:167 > No changes image=docker.io/linuxserver/letsencrypt:latest provider=docker,
Thu, 18 Jun 2020 16:12:05 EDT DBG app/internal/logging/logger.go:68 > [containers/image] Trying to access "docker.io/library/sha256:445909f0eaa51a79a49b1c4ae3f8b461cc4981df768f11d9f25c74573fe86f12",
Thu, 18 Jun 2020 16:12:05 EDT DBG app/internal/logging/logger.go:68 > [containers/image] Returning credentials from DockerAuthConfig,
Thu, 18 Jun 2020 16:12:05 EDT DBG app/internal/logging/logger.go:68 > [containers/image] Using registries.d directory /etc/containers/registries.d for sigstore configuration,
Thu, 18 Jun 2020 16:12:05 EDT DBG app/internal/logging/logger.go:68 > [containers/image]  No signature storage configuration found for docker.io/library/sha256:445909f0eaa51a79a49b1c4ae3f8b461cc4981df768f11d9f25c74573fe86f12,
Thu, 18 Jun 2020 16:12:05 EDT DBG app/internal/logging/logger.go:68 > [containers/image] Looking for TLS certificates and private keys in /etc/docker/certs.d/docker.io,
Thu, 18 Jun 2020 16:12:05 EDT DBG app/internal/logging/logger.go:68 > [containers/image] GET https://registry-1.docker.io/v2/,
Thu, 18 Jun 2020 16:12:05 EDT DBG app/internal/app/job.go:167 > No changes image=docker.io/nodered/node-red-docker:latest provider=docker,
Thu, 18 Jun 2020 16:12:05 EDT DBG app/internal/logging/logger.go:68 > [containers/image] Trying to access "docker.io/library/sha256:d918c177169ea239b3cffa74fa316255d69633054bc9f67e0a1ea11a58f3a1aa",
Thu, 18 Jun 2020 16:12:05 EDT DBG app/internal/logging/logger.go:68 > [containers/image] Returning credentials from DockerAuthConfig,
Thu, 18 Jun 2020 16:12:05 EDT DBG app/internal/logging/logger.go:68 > [containers/image] Using registries.d directory /etc/containers/registries.d for sigstore configuration,
Thu, 18 Jun 2020 16:12:05 EDT DBG app/internal/logging/logger.go:68 > [containers/image]  No signature storage configuration found for docker.io/library/sha256:d918c177169ea239b3cffa74fa316255d69633054bc9f67e0a1ea11a58f3a1aa,
Thu, 18 Jun 2020 16:12:05 EDT DBG app/internal/logging/logger.go:68 > [containers/image] Looking for TLS certificates and private keys in /etc/docker/certs.d/docker.io,
Thu, 18 Jun 2020 16:12:05 EDT DBG app/internal/logging/logger.go:68 > [containers/image] GET https://registry-1.docker.io/v2/,
Thu, 18 Jun 2020 16:12:05 EDT DBG app/internal/logging/logger.go:68 > [containers/image] Downloading /v2/linuxserver/radarr/blobs/sha256:c579beecd3dce402fe0c064c81511cea004ec8695ecbb605ae49b71d8dce75e4,
Thu, 18 Jun 2020 16:12:05 EDT DBG app/internal/logging/logger.go:68 > [containers/image] GET https://registry-1.docker.io/v2/linuxserver/radarr/blobs/sha256:c579beecd3dce402fe0c064c81511cea004ec8695ecbb605ae49b71d8dce75e4,
Thu, 18 Jun 2020 16:12:05 EDT DBG app/internal/logging/logger.go:68 > [containers/image] GET https://registry-1.docker.io/v2/sgtsquiggs/deluge-openvpn/manifests/latest,
Thu, 18 Jun 2020 16:12:05 EDT DBG app/internal/logging/logger.go:68 > [containers/image] Downloading /v2/linuxserver/piwigo/blobs/sha256:14d956ff1033ad033452255c230a094ad3efb90c03fbdc9fa3add74698f2462e,
Thu, 18 Jun 2020 16:12:05 EDT DBG app/internal/logging/logger.go:68 > [containers/image] GET https://registry-1.docker.io/v2/linuxserver/piwigo/blobs/sha256:14d956ff1033ad033452255c230a094ad3efb90c03fbdc9fa3add74698f2462e,
Thu, 18 Jun 2020 16:12:05 EDT DBG app/internal/logging/logger.go:68 > [containers/image] Downloading /v2/linuxserver/ombi/blobs/sha256:e8d1e54b8884ec760d7f7ea9ad16d919209e9424e9e218546643b9c4022e27bb,
Thu, 18 Jun 2020 16:12:05 EDT DBG app/internal/logging/logger.go:68 > [containers/image] GET https://registry-1.docker.io/v2/linuxserver/ombi/blobs/sha256:e8d1e54b8884ec760d7f7ea9ad16d919209e9424e9e218546643b9c4022e27bb,
Thu, 18 Jun 2020 16:12:05 EDT DBG app/internal/logging/logger.go:68 > [containers/image] Ping https://registry-1.docker.io/v2/ status 401,
Thu, 18 Jun 2020 16:12:05 EDT DBG app/internal/logging/logger.go:68 > [containers/image] GET https://auth.docker.io/token?scope=repository%3Alibrary%2Fsha256%3Apull&service=registry.docker.io,
Thu, 18 Jun 2020 16:12:05 EDT DBG app/internal/logging/logger.go:68 > [containers/image] Ping https://registry-1.docker.io/v2/ status 401,
Thu, 18 Jun 2020 16:12:05 EDT DBG app/internal/logging/logger.go:68 > [containers/image] GET https://auth.docker.io/token?scope=repository%3Alibrary%2Fsha256%3Apull&service=registry.docker.io,
Thu, 18 Jun 2020 16:12:05 EDT DBG app/internal/logging/logger.go:68 > [containers/image] Downloading /v2/linuxserver/sabnzbd/blobs/sha256:21db3434e75e8d8ea534b30aeb26dbe872d252af0a8c914308d60060cf417da0,
Thu, 18 Jun 2020 16:12:05 EDT DBG app/internal/logging/logger.go:68 > [containers/image] GET https://registry-1.docker.io/v2/linuxserver/sabnzbd/blobs/sha256:21db3434e75e8d8ea534b30aeb26dbe872d252af0a8c914308d60060cf417da0,
Thu, 18 Jun 2020 16:12:05 EDT DBG app/internal/logging/logger.go:68 > [containers/image] Downloading /v2/linuxserver/tautulli/blobs/sha256:a185fb280ae35511b708053c14c86e9a4413ffc73cfc088cc0565c4516a6201c,
Thu, 18 Jun 2020 16:12:05 EDT DBG app/internal/logging/logger.go:68 > [containers/image] GET https://registry-1.docker.io/v2/linuxserver/tautulli/blobs/sha256:a185fb280ae35511b708053c14c86e9a4413ffc73cfc088cc0565c4516a6201c,
Thu, 18 Jun 2020 16:12:05 EDT DBG app/internal/logging/logger.go:68 > [containers/image] GET https://registry-1.docker.io/v2/library/sha256/manifests/d918c177169ea239b3cffa74fa316255d69633054bc9f67e0a1ea11a58f3a1aa,
Thu, 18 Jun 2020 16:12:05 EDT DBG app/internal/logging/logger.go:68 > [containers/image] GET https://registry-1.docker.io/v2/library/sha256/manifests/445909f0eaa51a79a49b1c4ae3f8b461cc4981df768f11d9f25c74573fe86f12,
Thu, 18 Jun 2020 16:12:05 EDT DBG app/internal/logging/logger.go:68 > [containers/image] Downloading /v2/portainer/portainer/blobs/sha256:cd645f5a4769d8015c8b08bcf7164b4538c22982716565311f12c9b7455648f7,
Thu, 18 Jun 2020 16:12:05 EDT DBG app/internal/logging/logger.go:68 > [containers/image] GET https://registry-1.docker.io/v2/portainer/portainer/blobs/sha256:cd645f5a4769d8015c8b08bcf7164b4538c22982716565311f12c9b7455648f7,
Thu, 18 Jun 2020 16:12:05 EDT DBG app/internal/app/job.go:167 > No changes image=docker.io/linuxserver/radarr:5.14 provider=docker,
Thu, 18 Jun 2020 16:12:05 EDT DBG app/internal/logging/logger.go:68 > [containers/image] Trying to access "docker.io/library/telegraf:latest",
Thu, 18 Jun 2020 16:12:05 EDT DBG app/internal/logging/logger.go:68 > [containers/image] Returning credentials from DockerAuthConfig,
Thu, 18 Jun 2020 16:12:05 EDT DBG app/internal/logging/logger.go:68 > [containers/image] Using registries.d directory /etc/containers/registries.d for sigstore configuration,
Thu, 18 Jun 2020 16:12:05 EDT DBG app/internal/logging/logger.go:68 > [containers/image]  No signature storage configuration found for docker.io/library/telegraf:latest,
Thu, 18 Jun 2020 16:12:05 EDT DBG app/internal/logging/logger.go:68 > [containers/image] Downloading /v2/linuxserver/sonarr/blobs/sha256:c167287686687af9d408df23bf4be37921d49763e01b573e2374cf94828e3ac9,
Thu, 18 Jun 2020 16:12:05 EDT DBG app/internal/logging/logger.go:68 > [containers/image] GET https://registry-1.docker.io/v2/linuxserver/sonarr/blobs/sha256:c167287686687af9d408df23bf4be37921d49763e01b573e2374cf94828e3ac9,
Thu, 18 Jun 2020 16:12:05 EDT DBG app/internal/logging/logger.go:68 > [containers/image] Looking for TLS certificates and private keys in /etc/docker/certs.d/docker.io,
Thu, 18 Jun 2020 16:12:05 EDT DBG app/internal/logging/logger.go:68 > [containers/image] GET https://registry-1.docker.io/v2/,
Thu, 18 Jun 2020 16:12:05 EDT DBG app/internal/app/job.go:167 > No changes image=docker.io/linuxserver/ombi:latest provider=docker,
Thu, 18 Jun 2020 16:12:05 EDT DBG app/internal/app/job.go:167 > No changes image=docker.io/linuxserver/piwigo:latest provider=docker,
Thu, 18 Jun 2020 16:12:05 EDT DBG app/internal/app/job.go:167 > No changes image=docker.io/linuxserver/sabnzbd:latest provider=docker,
Thu, 18 Jun 2020 16:12:05 EDT DBG app/internal/logging/logger.go:68 > [containers/image] Ping https://registry-1.docker.io/v2/ status 401,
Thu, 18 Jun 2020 16:12:05 EDT DBG app/internal/logging/logger.go:68 > [containers/image] GET https://auth.docker.io/token?scope=repository%3Alibrary%2Ftelegraf%3Apull&service=registry.docker.io,
Thu, 18 Jun 2020 16:12:05 EDT DBG app/internal/app/job.go:167 > No changes image=docker.io/linuxserver/tautulli:latest provider=docker,
Thu, 18 Jun 2020 16:12:05 EDT DBG app/internal/app/job.go:167 > No changes image=docker.io/portainer/portainer:latest provider=docker,
Thu, 18 Jun 2020 16:12:05 EDT DBG app/internal/logging/logger.go:68 > [containers/image] GET https://registry-1.docker.io/v2/library/telegraf/manifests/latest,
Thu, 18 Jun 2020 16:12:05 EDT DBG app/internal/app/job.go:167 > No changes image=docker.io/linuxserver/sonarr:5.14 provider=docker,
Thu, 18 Jun 2020 16:12:05 EDT DBG app/internal/logging/logger.go:68 > [containers/image] GET https://registry-1.docker.io/v2/sgtsquiggs/deluge-openvpn/manifests/sha256:36e86a1f378cd9927e71e6603e85a93af7c3e805b1908c42774bae0fdcc9d04a,
Thu, 18 Jun 2020 16:12:05 EDT DBG app/internal/logging/logger.go:68 > [containers/image] Accessing "docker.io/library/sha256:d918c177169ea239b3cffa74fa316255d69633054bc9f67e0a1ea11a58f3a1aa" failed: Error reading manifest d918c177169ea239b3cffa74fa316255d69633054bc9f67e0a1ea11a58f3a1aa in docker.io/library/sha256: errors:,
denied: requested access to the resource is denied,
unauthorized: authentication required,
,
Thu, 18 Jun 2020 16:12:05 EDT ERR app/internal/app/diun.go:116 > Cannot run job error="Error reading manifest d918c177169ea239b3cffa74fa316255d69633054bc9f67e0a1ea11a58f3a1aa in docker.io/library/sha256: errors:\ndenied: requested access to the resource is denied\nunauthorized: authentication required\n" provider=docker,
Thu, 18 Jun 2020 16:12:05 EDT DBG app/internal/logging/logger.go:68 > [containers/image] Accessing "docker.io/library/sha256:445909f0eaa51a79a49b1c4ae3f8b461cc4981df768f11d9f25c74573fe86f12" failed: Error reading manifest 445909f0eaa51a79a49b1c4ae3f8b461cc4981df768f11d9f25c74573fe86f12 in docker.io/library/sha256: errors:,
denied: requested access to the resource is denied,
unauthorized: authentication required,
,
Thu, 18 Jun 2020 16:12:05 EDT ERR app/internal/app/diun.go:116 > Cannot run job error="Error reading manifest 445909f0eaa51a79a49b1c4ae3f8b461cc4981df768f11d9f25c74573fe86f12 in docker.io/library/sha256: errors:\ndenied: requested access to the resource is denied\nunauthorized: authentication required\n" provider=docker,
Thu, 18 Jun 2020 16:12:06 EDT DBG app/internal/logging/logger.go:68 > [containers/image] GET https://registry-1.docker.io/v2/library/telegraf/manifests/sha256:1c74ad11872b13eafe1e5f51fa46d00dab24a1b5610437d3c897ea3ba9b42b1a,
Thu, 18 Jun 2020 16:12:06 EDT DBG app/internal/logging/logger.go:68 > [containers/image] Downloading /v2/sgtsquiggs/deluge-openvpn/blobs/sha256:7fdacb7d9bda54cf2ff34f8a0a5f9eda7b3801d4e02afbc85bb7acbe5cc02fca,
Thu, 18 Jun 2020 16:12:06 EDT DBG app/internal/logging/logger.go:68 > [containers/image] GET https://registry-1.docker.io/v2/sgtsquiggs/deluge-openvpn/blobs/sha256:7fdacb7d9bda54cf2ff34f8a0a5f9eda7b3801d4e02afbc85bb7acbe5cc02fca,
Thu, 18 Jun 2020 16:12:06 EDT DBG app/internal/app/job.go:167 > No changes image=docker.io/sgtsquiggs/deluge-openvpn:latest provider=docker,
Thu, 18 Jun 2020 16:12:06 EDT DBG app/internal/logging/logger.go:68 > [containers/image] Downloading /v2/library/telegraf/blobs/sha256:ff4b325ac853bc8c7ae091dde59b698fdc060ecdffed2628a89b58602e1fff7f,
Thu, 18 Jun 2020 16:12:06 EDT DBG app/internal/logging/logger.go:68 > [containers/image] GET https://registry-1.docker.io/v2/library/telegraf/blobs/sha256:ff4b325ac853bc8c7ae091dde59b698fdc060ecdffed2628a89b58602e1fff7f,
Thu, 18 Jun 2020 16:12:06 EDT DBG app/internal/app/job.go:167 > No changes image=docker.io/library/telegraf:latest provider=docker,
Thu, 18 Jun 2020 16:12:06 EDT INF app/internal/app/diun.go:85 > Cron initialized with schedule 0 * * * *,
Thu, 18 Jun 2020 16:12:06 EDT INF app/internal/app/diun.go:89 > Next run in 47 minutes (2020-06-18 17:00:00 -0400 EDT),
```

**Comment by crazy-max** [comment]
Hi @cmartin616,

> Diun version : latest Docker image

I would need the semver release, latest can be anything. The exact version is displayed at the beginning of the logs.

> ```
> Thu, 18 Jun 2020 16:12:05 EDT ERR app/internal/app/diun.go:116 > Cannot run job error="Error reading manifest d918c177169ea239b3cffa74fa316255d69633054bc9f67e0a1ea11a58f3a1aa in docker.io/library/sha256: errors:\ndenied: requested access to the resource is denied\nunauthorized: authentication required\n" provider=docker,
> Thu, 18 Jun 2020 16:12:05 EDT DBG app/internal/logging/logger.go:68 > [containers/image] Accessing "docker.io/library/sha256:445909f0eaa51a79a49b1c4ae3f8b461cc4981df768f11d9f25c74573fe86f12" failed: Error reading manifest 445909f0eaa51a79a49b1c4ae3f8b461cc4981df768f11d9f25c74573fe86f12 in docker.io/library/sha256: errors:,
> denied: requested access to the resource is denied,
> unauthorized: authentication required,
> ,
> ```

Ok it looks like one of the images you watch in your Docker instance is now private. Hard to find the name of the image with the manifest hash. Can you list here the images used by your containers please? You can use the following command to print them `docker container ls --format "{{.Image}}"`

**Comment by cmartin616** [comment]
Thanks for the quick reply.  Here is the information you requested but a solution at the end.

Very first line of the log:
Thu, 18 Jun 2020 16:12:02 EDT INF app/cmd/main.go:49 > Starting Diun 3.0.0

Results of container list (name added if image name is not clear):
crazymax/diun:latest
linuxserver/sonarr:5.14
linuxserver/ombi:latest
linuxserver/ddclient
sgtsquiggs/deluge-openvpn
homeassistant/home-assistant
linuxserver/radarr:5.14
mariadb
linuxserver/heimdall
linuxserver/sabnzbd
linuxserver/calibre-web
housewrecker/gaps:latest
d918c177169e (jackett)
linuxserver/lazylibrarian
linuxserver/letsencrypt
influxdb
telegraf
haveagitgat/tdarr
linuxserver/tautulli
grafana/grafana
linuxserver/piwigo
portainer/portainer
445909f0eaa5 (bazarr)
nodered/node-red-docker

----
I did some digging based on your private image comment.  It turns out some of my containers started with the image name set to a sub-string of the sha256 ID.  I don't know why this happened but destroying the container and recreating it with the same docker command resulted in a correct image name.  Diun started sending notifications after these bad image names were removed.  Interestingly enough, several images had updates but it seems this error caused the process to stop checking and/or notifying as I did not receive notifications on the other 'good' image names until I fixed the problematic ones.

This is probably user error and can be avoided in the future with a bit more knowledge.  Any information about why this might have happened would be appreciated.  

As a result of this issue, maybe it would be worthwhile to include some error handling so diun sends a notification for a private image so users know their image isn't able to be watched any further?

Thanks again for your help.

**Comment by crazy-max** [comment]
@cmartin616 

> I did some digging based on your private image comment. It turns out some of my containers started with the image name set to a sub-string of the sha256 ID.

I see `445909f0eaa5 (bazarr)` and `d918c177169e (jackett)` were the culprits here. Can you show me your compose or run command for those ones?

> Diun started sending notifications after these bad image names were removed. Interestingly enough, several images had updates but it seems this error caused the process to stop checking and/or notifying as I did not receive notifications on the other 'good' image names until I fixed the problematic ones.

I think we could improve the process here and continue on error. Will check this out.

> This is probably user error and can be avoided in the future with a bit more knowledge. Any information about why this might have happened would be appreciated.

Do you use any kind of software to update your image's container?

> As a result of this issue, maybe it would be worthwhile to include some error handling so diun sends a notification for a private image so users know their image isn't able to be watched any further?

Sounds good to me.

**Comment by cmartin616** [comment]
I run all my docker create commands as *.sh because I can't figure out the correct way to use Docker compose and get control of the shell back when I'm done.  I'm often destroying, updating and creating multiple containers in series so its a pain to terminate the SSH session each time.  I also probably have an issue with tags as I don't understand them well.  My goal is to keep the container updated to the image with the latest version of the software.  Below are my Jackett and Bazarr configs:

docker create \
  --name=jackett \
  -e PUID=[uid]\
  -e PGID=[gid] \
  -e TZ=America/New_York \
  -p 9117:9117 \
  -v /opt/jackett:/config \
  --restart unless-stopped \
  linuxserver/jackett:latest

docker create \
  --name=bazarr \
  -e PUID=[uid] \
  -e PGID=[gid]\
  -e TZ=America/New_York \
  -p 6767:6767 \
  -v /opt/bazarr:/config \
  -v /mnt/nas/movies:/movies \
  -v /mnt/nas/tv:/tv \
  --restart unless-stopped \
  linuxserver/bazarr

The only things I do to update the containers are these steps:

1. Notification from duin
2. docker stop [container]
3. docker rm [container]
4. path to [container].sh to create
5. docker start [container]

I'm definitely open to better methods to manage this if I am making mistakes.  Thank you for the help and very useful script!

**Comment by crazy-max** [comment]
@cmartin616 

> `docker create ...`

Ok now I know why you have this kind of issue. You should use `docker run` if you want to run a container. `create` should not be used in your case because it creates a writeable container layer over the specified image. That's why it resolves with a hash. I will handle this case in Diun.

**Comment by crazy-max** [comment]
@cmartin616 

> the previous container still exist and the old image will be in "dangling" state. That's why it resolves with a hash. I will handle this case in Diun.

Dangling images are now skipped on 4.0.0-beta.4

**Comment by crazy-max** [comment]
> Dangling images are now skipped on 4.0.0-beta.4

Available through 4.0.0 release. See https://crazy-max.github.io/diun/changelog/#400-20200622.

**Comment by cmartin616** [comment]
Thank you so much!
